### ISSUE TITLE
Ban impossible requests

### ISSUE THREAD

**Issue by mm201** [mm201/pkmnFoundations#4]
Certain legendary / non-breedable Pokémon can't be obtained at Lv. 9 and under. Troublemakers often ask for these so they can show off their Pokémon without them being traded. (Hackers inevitably take them anyway.) Rather than letting them gum up our system, we should tell them they can't ask for that in a friendly way.

**Comment by PokeAcer549** [comment]
So we need to make the pokemonpds script do a 404 not found if they try (I can do a ton of 'under 9 legendaries' put ups and retrieval for the code they send) which DC's them and then we put up a notice saying this will DC you.

**Comment by mm201** [comment]
No. There's a proper error code which will make the game say "You can't ask for that".

I could throw together a hackish implementation very quickly but I want to do it right and base it off encounter tables. (including events)

**Comment by PokeAcer549** [comment]
Ah. But you will be able to pull the pokemon back out if someone does it before the update right?

----- Reply message -----

**Comment by PokeAcer549** [comment]
Project Pokemon has GTS protocol documented, and here's what it says.
"Pokémon are offered on the GTS by requesting /pokemondpds/worldexchange/post.asp. If the Pokémon is rejected by the server, the response is 0x000c; otherwise, if the deposit is successful, 0x0001."

Source: http://projectpokemon.org/wiki/GTS_protocol

**Comment by mm201** [comment]
That's the easy part. (IIRC I have a comment in that area of the code documenting responses.) The hard part is validating movesets and doing it the right way.

**Comment by ChynenseFuud** [comment]
Recently some guy requested a Rattata for a Meloetta in the DPPHGSS section in the server, is there something that could be done?, because as far as I know there is no way to obtain a Meloetta in the 4th generation.
![ohgod](https://user-images.githubusercontent.com/43972131/46634094-2ed97880-cb26-11e8-8c32-dd67ee39478a.png)

**Comment by zurgeg** [comment]
> 
> 
> No. There's a proper error code which will make the game say "You can't ask for that".
> 
> I could throw together a hackish implementation very quickly but I want to do it right and base it off encounter tables. (including events)

PKHeX (and likely it's Nuget version) supports looking up encounters, which might come in handy

**Comment by zurgeg** [comment]
> 
> 
> Recently some guy requested a Rattata for a Meloetta in the DPPHGSS section in the server, is there something that could be done?, because as far as I know there is no way to obtain a Meloetta in the 4th generation. ![ohgod](https://user-images.githubusercontent.com/43972131/46634094-2ed97880-cb26-11e8-8c32-dd67ee39478a.png)

Correct, additionally, Meloetta can't be caught in a master ball either iirc

**Comment by mm201** [comment]
These things have been blocked for months now. And this thread is about impossible requests, e.g. Kyogre 9 and under. You're looking for #3.

**Comment by mm201** [comment]
A flabbergasted expression of frustration. I have updated the description of this 7 year old todo issue to hopefully avoid confusion.

**Comment by mm201** [comment]
So it appears that 'you can't ask for that' only happens on the client and the only things I can make it say are "[That Pokémon may not be offered for trade!](https://github.com/mm201/pkmn-classic-framework/blob/master/gts/pokemondpds.ashx.cs#L290)" and a few error messages. I may close this if I can't find a workaround.
